File: JavascriptAlgorithm/Array/maxArea-011.js
var maxArea = function (height) {
  const length = height.length;
  let maxArea = 0;
  for (let l = 0, r = length - 1; l < r; ) {
    let area =
      height[l] > height[r]
        ? height[r--] * (r - l + 1)
        : height[l++] * (r - l + 1);
    maxArea = Math.max(area, maxArea);
  }
  return maxArea;
};
export {maxArea};

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/ThirdRound/Chapter03/maxArea_011.java
package algo.tzashinorpu.ThirdRound.Chapter03;

public class maxArea_011 {
    public int maxArea(int[] height) {
        int max = 0;
        for (int i = 0, j = height.length - 1; i < j; ) {
            int h = height[i] < height[j] ? height[i++] : height[j--];
            max = Math.max(max, h * ((j - i) + 1));
        }
        return max;
    }
}
